### ISSUE TITLE
The progress bar for `merge_datasets` does not accurately represent the status of the job.

### ISSUE THREAD

**Issue by abysmalocean** [Lightning-AI/litdata#433]
## 🐛 Bug

The progress bar for `merge_datasets` does not accurately represent the status of the job.
The progress bar is only report submit the job, not finishing status. 

original code : 
```py
with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
        futures: List[concurrent.futures.Future] = []
        for copy_info in _tqdm(copy_infos):
            future = executor.submit(_apply_copy, copy_info, resolved_output_dir)
            futures.append(future)
```

in this case, the progress bar only reports the status of job submission.

Change to the following code, will solve the issue.  

```py
with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
        futures: List[concurrent.futures.Future] = []
        for copy_info in copy_infos:
            future = executor.submit(_apply_copy, copy_info, resolved_output_dir)
            futures.append(future)

        for future in _tqdm(concurrent.futures.as_completed(futures), total=len(futures)):
            future.result()
``` 

### To Reproduce

Steps to reproduce the behavior...

<details>
  <summary>Code sample</summary>

```python
# Ideally attach a minimal code sample to reproduce the decried issue.
# Minimal means having the shortest code but still preserving the bug.
```

</details>

### Additional context

<details>
  <summary>Environment detail</summary>

- PyTorch Version (e.g., 1.0):
- OS (e.g., Linux):
- How you installed PyTorch (`conda`, `pip`, source):
- Build command you used (if compiling from source):
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- Any other relevant information:

</details>

**Comment by tchaton** [comment]
Hey @abysmalocean. Great catch ! Would you mind making a PR to fix it ?

**Comment by abysmalocean** [comment]
yes, I could.  would be my first PR. I would like to contribute my effort.

**Comment by tchaton** [comment]
Yeah @abysmalocean Super excited for your first PR and feel free to continue after with issues, feature requests or even new PRs.

**Comment by bhimrazy** [comment]
Hi @abysmalocean,  
Just checking in to see if there’s any update on the PR for this issue. 
Let us know if you need any help or have any blockers!
